### ISSUE TITLE
Veracode warning on 'product-agents-edc'

### ISSUE THREAD

**Issue by drcgjung** [eclipse-tractusx/sig-infra#65]
Please report **undisclosed** or **confidential** vulnerabilities here: https://www.eclipse.org/security/

**Topics (Please mark an [x] to your Topic):**
- [ ] KICS
- [ ] Invicti
- [ ] GitGuardian
- [ ] Pentesting
- [ ] Security assessment (threat modeling, code reviews)
- [ ] Trivy
- [x] Veracode (initial setup, GitHub integration)
- [ ] Other

**Please describe your request:**

For 'product-agents-edc' we now successfully setup the veracode workflow (adopted from 'tx-tractusx-edc' https://github.com/catenax-ng/tx-tractusx-edc/blob/main/.github/workflows/veracode.yaml)

https://github.com/catenax-ng/product-agents-edc/actions/runs/5195436041

But the uploading of the artifact jars seems to run into a warning, e.g.

[23.06.07 02:49:50] 
[23.06.07 02:49:50] Application profile "knowledge-agents-edc/controlplane-memory" (appid=1748166) was located.
[23.06.07 02:49:50] 
[23.06.07 02:49:50] Attempting to create a new build with name "controlplane-memory-c5511f6e17788b497f14b3[27](https://github.com/catenax-ng/product-agents-edc/actions/runs/5195436041/jobs/9368152181#step:7:28)1d67112d05a92ff8".
[23.06.07 02:49:51] 
[23.06.07 02:49:51] * Action "UploadAndScan" returned the following message:
[23.06.07 02:49:51] * App not in state where new builds are allowed.
[23.06.07 02:49:51] 
[23.06.07 02:49:52] 
[23.06.07 02:49:52] * A scan is in progress or has failed to complete successfully. Wait for the current scan to complete or delete the failed scan from the Veracode Platform and try again.

Is that because the application profile name does not match/prefix with the 'product-agents-edc' name or do we have to setup some additional things there?

**Comment by the-tatanka** [comment]
![image](https://github.com/eclipse-tractusx/sig-infra/assets/59142915/3abc48f1-f171-437b-9dc4-9e888b084f61)

Hi @drcgjung , Veracode can't scan your uploaded files. Please check your uploaded files.
The projects at the bottom of the picture have the same problem.

Do you need all these projects? If no, please let me know - I will remove them.

Removed all running scans (also knowledge-agents-edc/controlplane-memory" (appid=1748166)), you can retest. Means, trigger your Veracode pipeline first and then see which jobs still go wrong.

![image](https://github.com/eclipse-tractusx/sig-infra/assets/59142915/4a112d35-a9e3-4e72-b7b4-66c21dba18cd)

**Comment by FaGru3n** [comment]
@drcgjung Please review, the comment from @the-tatanka 

> ![image](https://user-images.githubusercontent.com/59142915/243969803-3abc48f1-f171-437b-9dc4-9e888b084f61.png)
> 
> Hi @drcgjung , Veracode can't scan your uploaded files. Please check your uploaded files. The projects at the bottom of the picture have the same problem.
> 
> Do you need all these projects? If no, please let me know - I will remove them.
> 
> Removed all running scans (also knowledge-agents-edc/controlplane-memory" (appid=1748166)), you can retest. Means, trigger your Veracode pipeline first and then see which jobs still go wrong.
> 
> ![image](https://user-images.githubusercontent.com/59142915/243970846-4a112d35-a9e3-4e72-b7b4-66c21dba18cd.png)

**Comment by drcgjung** [comment]
Excuse me for the late answer, was on holiday.

Since I have not worked with veracode before, It was more of a copy&paste thing from tractusx-edc
(see for example https://github.com/eclipse-tractusx/tractusx-edc/actions/runs/5238983165/jobs/9458401715)

The repo results in 6 docker images (4 extended control planes, 2 additional data planes)
I presumed that we need to setup a veracode module per resulting image.

What I did not notice (until now) was that tractusx-edc moved from a classpath/lib-based approach to an uber-jar approach in the 0.4 versions. So it is clear now that veracode cannot find an endpoint in the target jar. 

Should I rather:
- add the classpath libs to the tgz as well or rather
- move to an uber-jar approach (and put several uber-jars in a single veracode module tgz)

Thx for caring,
CGJ
